model.js

```diff
@@ -43,4 +46,19 @@ const fetchComments = (reviewId) => {
    })
 }
 
-module.exports = {fetchCategories, fetchReviews, fetchReviewById, fetchComments}
\ No newline at end of file
+const newComment = (review_id, comment) => {
+   const {body, username} = comment
+   return db.query(`INSERT INTO comments (body, review_id, author) VALUES ($1,$2,$3) RETURNING *`, [body, review_id, username])
+   .then((comment)=>{
+      return {comment: comment.rows}
+   })
+   .catch(()=>{
+      let accepted = +review_id
+      if(!accepted){
+         return Promise.reject({status:400, msg: "Invalid Id, please enter a number."})
+      }
+      return Promise.reject({status:404, msg: "You can't comment on a review that doesn't exist!"})
+   })
```

Comment:
as you have a .catch in your model and your controller you now have multiple .catches in your promise chain. you can reject your promise in your model but you don't need a catch block here as your controller will catch the error.